test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -324,6 +326,11 @@ namespace integration_framework {
         ->onExpiredBatches();
   }
 
+  rxcpp::observable<iroha::network::Commit>
+  IntegrationTestFramework::getYacOnCommitObservable() {
+    return iroha_instance_->getIrohaInstance()->getConsensusGate()->on_commit();
```

Comment:
Maybe it's better use push to the queue instead of sharing observable at the interface?

Reply:
I think observables are more flexible as they allow (for example) ignoring, filtering and tee'ing objects, and much more.